sql/core/src/test/scala/org/apache/spark/sql/connector/DataSourceV2SQLSuite.scala

```diff
@@ -830,6 +830,70 @@ class DataSourceV2SQLSuite
     assert(df.collect().map(_.getAs[String](0)).sorted === expected.sorted)
   }
 
+  test("UseCatalog: use catalog with v2 catalog") {
+    val catalogManager = spark.sessionState.catalogManager
+    assert(catalogManager.currentCatalog.name() == "session")
+
+    sql("USE CATALOG testcat")
+    assert(catalogManager.currentCatalog.name() == "testcat")
+  }
+
+  test("UseCatalog: v2 catalog does not exist") {
+    val exception = intercept[AnalysisException] {
+      sql("USE CATALOG unknown")
+    }
+    assert(exception.getMessage.contains("v2 catalog 'unknown' cannot be loaded"))
+  }
+
+  test("Use: basic tests with USE statements") {
+    val catalogManager = spark.sessionState.catalogManager
+
+    // Validate the initial current catalog and namespace.
+    assert(catalogManager.currentCatalog.name() == "session")
+    assert(catalogManager.currentNamespace === Array("default"))
+
+    // The following implicitly creates namespaces.
+    sql("CREATE TABLE testcat.ns1.ns1_1.table (id bigint) USING foo")
+    sql("CREATE TABLE testcat2.ns2.ns2_2.table (id bigint) USING foo")
+    sql("CREATE TABLE testcat2.ns3.ns3_3.table (id bigint) USING foo")
+
+    // Catalog is explicitly specified as 'testcat'.
+    sql("USE ns1.ns1_1 IN testcat")
+    assert(catalogManager.currentCatalog.name() == "testcat")
+    assert(catalogManager.currentNamespace === Array("ns1", "ns1_1"))
+
+    // Catalog is resolved to 'testcat2'.
+    sql("USE testcat2.ns2.ns2_2")
```

Comment:
If we support this, maybe it's better to have 3 commands
1. USE CATALOG ...
2. USE NAMESPACE ...
3. USE [catalogName].ns1.ns2...

Since the catalog has the highest priority during name resolution, there is no ambiguity to switch to a specific namespace in a specific catalog, via `USE catalogName.ns1.ns2`. That said, we don't need `USE ... IN ...`

Reply:
If we go with this proposal, we can simplify the parser a little bit
```
USE (CATALOG | NAMESPACE)? multipartIdentifier
```

```
def visitUse(ctx: UseContext): LogicalPlan = withOrigin(ctx) {
  val nameParts = visitMultipartIdentifier(ctx.multipartIdentifier)
  // USE CATALOG can only specify a single catalog name
  if (ctx.CATALOG != null && nameParts.length > 1) parser error
  UseStatement(ctx.CATALOG != null, ctx.NAMESPACE != null, nameParts)  
}
...
case class UseStatement(isCatalog: Boolean, isNamespace: Boolean, nameParts: Seq[String]) {
  if (isCatalog) {
    assert(!isNamespace) // isCatalog and isNamespace can't be both true
    assert(nameParts.length ==1) // USE CATALOG can only specify a single catalog name
  }
}
...

```

Reply:
I am fine with this proposal, but do we even need `2) USE NAMESPACE ...`?

Reply:
image you have a namepace `abc` in the current catalog, and a catalog called `abc`, `USE NAMESPACE abc` can set current namespace to `abc` without ambiguity.

Actually this reminds me that we don't need `USE CATALOG` as `USE catalogName` can set current catalog without ambiguity. So we can simplify it a bit more:
```
USE NAMESPACE? multipartIdentifier
```
```
def visitUse(ctx: UseContext): LogicalPlan = withOrigin(ctx) {
  val nameParts = visitMultipartIdentifier(ctx.multipartIdentifier)
  UseStatement(ctx.NAMESPACE != null, nameParts)
}
...
case class UseStatement(isSetNamespace: Boolean, nameParts: Seq[String])
```

And the rule is simple:
1. USE NAMESPACE only sets the current namespace
2. USE sets the current catalog if the first name part is a catalog name. This doesn't change even if there is a namespace with the same name. Catalog has the highest priority during name resolution.

Reply:
OK. Sounds good to me. Let me try this.